tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/DefaultTaggedMetricRegistry.java

```diff
@@ -123,6 +123,12 @@ public void addMetrics(String safeTagName, String safeTagValue, TaggedMetricSet
         return Optional.ofNullable(taggedRegistries.remove(Maps.immutableEntry(safeTagName, safeTagValue)));
     }
 
+    @Override
+    public boolean removeMetrics(
+            String safeTagName, String safeTagValue, TaggedMetricSet metrics) {
+        return taggedRegistries.remove(Maps.immutableEntry(safeTagName, safeTagValue), metrics);
```

Comment:
Do we need to implement `equals`/`hashCode` on `TaggedMetricSet` impls?

Reply:
I don't think so, it's the same contract as Map#remove. The point is you provide the same instance to remove as to add, so it's up to you.

Reply:
Basically this is fixing the fact that there isn't really a safe way to remove and then add if you can't guarantee the ordering; like if you have a cache of some objects that produce the metrics, and upon deleting from the cache you unregister the taggedmetrics, because usually cache removal notification is asynchronous, you can't guarantee that the object hasn't been recreated already.

Reply:
makes sense, let's just document the contract in the interface and should be good to go